openmc/deplete/results_list.py

```diff
@@ -0,0 +1,167 @@
+import h5py
+import numpy as np
+
+from .results import Results, _VERSION_RESULTS
+from openmc.checkvalue import check_filetype_version
+
+
+__all__ = ["ResultsList"]
+
+
+class ResultsList(list):
+    """A list of openmc.deplete.Results objects
+
+    It is recommended to use :meth:`from_hdf5` over
+    direct creation.
+    """
+
+    @classmethod
+    def from_hdf5(cls, filename):
+        """Load in depletion results from a previous file
+
+        Parameters
+        ----------
+        filename : str
+            Path to depletion result file
+
+        Returns
+        -------
+        new : ResultsList
+            New instance of depletion results
+        """
+        with h5py.File(str(filename), "r") as fh:
+            check_filetype_version(fh, 'depletion results', _VERSION_RESULTS[0])
+            new = cls()
+
+            # Get number of results stored
+            n = fh["number"][...].shape[0]
+
+            for i in range(n):
+                new.append(Results.from_hdf5(fh, i))
+        return new
+
+    def get_atoms(self, mat, nuc):
+        """Get number of nuclides over time from a single material
+
+        .. note::
+
+            Initial values for some isotopes that do not appear in
+            initial concentrations may be non-zero, depending on the
+            value of :class:`openmc.deplete.Operator` ``dilute_initial``.
+            The :class:`openmc.deplete.Operator` adds isotopes according
+            to this setting, which can be set to zero.
+
+        Parameters
+        ----------
+        mat : str
+            Material name to evaluate
+        nuc : str
+            Nuclide name to evaluate
+
+        Returns
+        -------
+        time : numpy.ndarray
+            Array of times in [s]
+        concentration : numpy.ndarray
+            Total number of atoms for specified nuclide
+
+        """
+        time = np.empty_like(self, dtype=float)
```

Comment:
`time` is the name of a standard Python library module: https://docs.python.org/3/library/time.html Consider using a different variable name. Also, this value seems to be used like an array, so maybe `times` or `result_times` are appropriate.